src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -544,7 +544,9 @@ public function isRelation($key)
         if (method_exists($this, $key)) {
             $relation = $this->$key();
 
-            return $relation instanceof Relation;
+            if ($relation instanceof Relation) {
```

Comment:
i meant not calling the method but instead doing it over the reflection api like described.